### PR TITLE
refactor(metadata): use same host name lookup as in sitemap.xml

### LINKED ISSUE #284: URLs in open graph et al.
currently the og meta data looks something like this:

```
 <link rel="canonical" href="https://design--theblog--adobe.hlx.page/en/drafts/migrated/2020/04/30/protect-your-most-sensitive-digital-assets-while-working-from-home.html">
  <meta property="og:url" content="https://design--theblog--adobe.hlx.page/en/drafts/migrated/2020/04/30/protect-your-most-sensitive-digital-assets-while-working-from-home.html">
  <meta property="og:image" content="https://hlx.blob.core.windows.net/external/e6db6c540a6c693f17b4611ffff718968a71f2f7">
  <meta property="og:image:secure_url" content="https://hlx.blob.core.windows.net/external/e6db6c540a6c693f17b4611ffff718968a71f2f7">
  
  <meta name="twitter:image" content="https://hlx.blob.core.windows.net/external/e6db6c540a6c693f17b4611ffff718968a71f2f7">
```

i think that all of the URLs are misleading, most importantly the ones that show the canonical URLs...
ideally they should be something like:

```
 <link rel="canonical" href="https://blog.adobe.com/en/drafts/migrated/2020/04/30/protect-your-most-sensitive-digital-assets-while-working-from-home.html">
  <meta property="og:url" content="https://blog.adobe.com/en/drafts/migrated/2020/04/30/protect-your-most-sensitive-digital-assets-while-working-from-home.html">
  <meta property="og:image" content="https://blog.adobe.coml/hlx_e6db6c540a6c693f17b4611ffff718968a71f2f7">
  <meta property="og:image:secure_url" content="https://blog.adobe.com/hlx_e6db6c540a6c693f17b4611ffff718968a71f2f7">
  
  <meta name="twitter:image" content="https://blog.adobe.com/hlx_e6db6c540a6c693f17b4611ffff718968a71f2f7">

```

i appreciate the complexity with that and wonder what the impact is of just using URLs relative to the host... eg...

```
 <link rel="canonical" href="/en/drafts/migrated/2020/04/30/protect-your-most-sensitive-digital-assets-while-working-from-home.html">

```

i found some conflicting documentation on that but i am not sure if that even works for the `og:` and `twitter:` meta data

### PR DESCRIPTION
**Pull request by rofe**
FIx #284

### CODE REVIEW

File: cgi-bin/sitemap.js [hunk 1]
```diff
@@ -12,9 +12,9 @@
 const algoliasearch = require('algoliasearch');
 const { logger } = require('@adobe/openwhisk-action-logger');
 const { wrap } = require('@adobe/openwhisk-action-utils');
-
 const Downloader = require('@adobe/helix-pipeline/src/utils/Downloader.js');
 const { MountConfig, IndexConfig } = require('@adobe/helix-shared');
+const { getOriginalHost } = require('../src/utils');
```
Comment (dominique-pfister) [inline cgi-bin/sitemap.js:17]: I sometimes see `require('../src/file.js')` and somtimes `require('../src/file')`, which are both working: do we have a convention on this?

File: src/html.pre.js [hunk 2]
```diff
@@ -80,7 +81,7 @@ function pre(context) {
     .toArray();
   meta.description = `${desc.slice(0, 25).join(' ')}${desc.length > 25 ? ' ...' : ''}`;
   // url: use outer CDN URL if possible
-  meta.url = request.headers['x-cdn-url'] || `https://${request.headers.host}${request.url}`;
+  meta.url = `https://${getOriginalHost(request.headers)}${request.url}`;
```
Comment (dominique-pfister) [inline src/html.pre.js:84]: The `request.headers['x-cdn-url']` preference is now missing, is that deliberate?

Comment (rofe) [reply src/html.pre.js:84]: Yes 🙂 it should now use the exact same headers for the host name as the sitemap.xml.

File: cgi-bin/sitemap.js (hunk 1 above)
Comment (rofe) [reply cgi-bin/sitemap.js:17]: I'm not sure, I usually omit the extension. Must have seen it elsewhere 😄

**Pull request merged**